src/python/tritonfrontend/_api/validation.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from typing import Any
+
+
+class Validation:
+    def validate_type(self, value: Any, expected_type: Any, param_name: str):
+        if not isinstance(value, expected_type):
+            raise TypeError(
+                f"Incorrect Type for {param_name}. Expected {expected_type}, got {type(value)}"
+            )
+
+    # TODO: implement to catch ints in python that are too big for int32_t in C++
+    # TODO: Make sure port >= 0. Cannot have negative port.
+    def validate_range(self, value, lb, ub, param_name):
+        pass
+
+    def validate(self):
```

Comment:
Have you looked into `pydantic`? https://docs.pydantic.dev/latest/concepts/fields/#numeric-constraints

Reply:
Thanks for suggestion. I converted KServeGrpc.Options and KServeHttp.Options to use `@pydantic.dataclasses.dataclass`. Without using the `Validation` class, I get the following error messages:
![image](https://github.com/user-attachments/assets/b5024df8-e2d5-48c1-909c-79c83ffb8756)
![image](https://github.com/user-attachments/assets/28033847-bf82-426f-a04e-a27615dac491)